traitsui/qt4/ui_panel.py

```diff
@@ -1310,7 +1310,7 @@ def __init__(self, parent, html, scale_dx, scale_dy):
 
         # Position and show the dialog
         position_window(self, parent=parent)
-        self.show()
+        self.exec_()
```

Comment:
I'd like this PR merged by the end of this week so if we haven't figured out how to add a regression test by then, let's just merge this PR and open an issue to write a regression test.